src/50_Samples_%26_Templates/Product.html

```diff
@@ -278,53 +346,177 @@ <h4 class="category"><a href="/samples_templates/product_listing/preview/">More<
+    <!-- We are using [amp-bind](https://www.ampproject.org/docs/reference/components/amp-bind) to update the price of the product
+      depending on the selected colour. -->
+    <!--We bind to the `text` attribute: the value of text is the
+      value of expression `product[selectedColor].price`. When this expression is evaluated,
+      it gets the price value from the json inside the `amp-state`.-->
+    <!--Notice how we set a default value for the price: this is because the value defined inside `amp-state`
+      will not be accessed at the page load. We set `green` and `S` as default color and default size
+      as the correspondent variables have null value at default.-->
+    <!--As soon as the state of this variable changes, the expression is evaluated.
+      See below to understand how we change the state of the 2 variables `selectedColor` and `selectedSize`. -->
+      <p class="price-description">Price: <span [text]="product[selectedColor || 'green'].sizes[selectedSize || 'S']">$1.99</span>
+    </p>
+    <!-- The add to cart action is implemented using  [amp-form](/components/amp-form/). In our sample we use the [amp-selector](/components/amp-selector/) for selecting different product properties.
+    Pressing the `ADD TO CART` button adds the product to a shopping cart page using the properties you have selected. Notice that the button URL contains the query `clientId={{ClientId}}`.
+    If an item is successfully added to the shopping cart, we redirect the user to the shopping cart page.
+    The redirect target is configured via a header in the server's form response ([redirecting after submission](https://github.com/ampproject/amphtml/blob/master/extensions/amp-form/amp-form.md#redirecting-after-a-submission)):
 
       access-control-expose-headers:AMP-Access-Control-Allow-Source-Origin,AMP-Redirect-To
-      amp-redirect-to:http://ampbyexample.com/shopping_cart/?clientid=amp-123456789
+      amp-redirect-to:http://ampbyexample.com/shopping_cart/?clientid=amp-123456789 -->
 
-    We are using the `CLIENT_ID` variable to identify the user, this enables storing a shopping cart between repeated visits of an AMP page (either via the AMP Cache or the original host). This variable can be used inside an amp-form by declaring an hidden input value with `default-value="CLIENT_ID(cart)`. Read more about variable substitution [here](https://github.com/ampproject/amphtml/blob/master/spec/amp-var-substitutions.md). -->
+      <!-- We are using `amp-selector` together with `amp-bind` to update the page for showing the price, the size availability, the gallery and preview of images.
```

Comment:
This reads weird at this point - can we move it earlier to the other part where we explain amp-bind?